backend/core/serializers/user.py

```diff
@@ -21,26 +22,53 @@ def _validate_username(value):
     return value
 
 
+@wrap_validation_error_with_field("email")
+def validate_email_uniqueness_iexact(value: str) -> str:
+    """Raise an error if an email is considered already in use.
+
+    Even if emails are considered case-sensitive by the RFC 5321, several
+    email providers don't follow this specification. This validator ensures
+    users won't be able to create several accounts with the same address
+    by changing only the case.
+    """
+    try:
+        User.objects.get(email__iexact=value.lower())
+    except User.DoesNotExist:
+        pass
+    else:
+        raise ValidationError(_("A user with this email address already exists."))
+
+    return value
+
+
 class RegisterUserSerializer(DefaultRegisterUserSerializer):
     def validate_username(self, value):
         return _validate_username(value)
 
+    def validate(self, data):
+        validate_email_uniqueness_iexact(data["email"])
+        super(RegisterUserSerializer, self).validate(data)
+        return data
+
 
 class RegisterEmailSerializer(DefaultRegisterEmailSerializer):
-    email = EmailField(validators=[
-        UniqueValidator(
-            queryset=User.objects.all(),
-            message="A user with this email address already exists."
-        ),
-    ])
+    email = EmailField(
+        validators=[
+            UniqueValidator(
+                queryset=User.objects.all(),
+                message=_("A user with this email address already exists."),
+                lookup="iexact",
+            ),
+        ]
+    )
```

Comment:
Cannot the same field definition be used in the `RegisterUserSerializer`?  
It would make the maintenance easier to use the same mechanism in both.

Reply:
Sure!